src/ape/api/accounts.py

```diff
@@ -1,81 +1,96 @@
+    def sign_transaction(self, txn: TransactionAPI) -> TransactionAPI:
+        # NOTE: Some accounts may not offer signing things
+        return txn
 
-    def __str__(self) -> str:
-        return self.address
+    def transfer(self, account: "AddressAPI", value: int = None, data: bytes = None) -> ReceiptAPI:
+        txn = self._transaction_class(  # type: ignore
+            sender=self.address,
+            receiver=account.address,
+            nonce=self.nonce,
+        )
 
-    @property
-    def nonce(self) -> int:
-        return self._provider.get_nonce(self.address)
+        if data:
+            txn.data = data
 
-    @property
-    def balance(self) -> int:
-        return self._provider.get_balance(self.address)
+        if value:
+            txn.value = value
 
-    @property
-    def code(self) -> bytes:
-        # TODO: Explore caching this (based on `self.provider.network` and examining code)
-        return self._provider.get_code(self.address)
+        else:
+            # NOTE: If `value` is `None`, send everything
+            txn.value = self.balance - txn.gas_limit * txn.gas_price
 
-    @property
-    def codesize(self) -> int:
-        return len(self.code)
+        return self.call(txn)
 
-    @property
-    def is_contract(self) -> bool:
-        return len(self.code) > 0
+    def call(self, txn: TransactionAPI) -> ReceiptAPI:
+        txn.gas_limit = self._active_provider.estimate_gas_cost(txn)
+        txn.gas_price = self._active_provider.gas_price
 
+        if txn.gas_limit * txn.gas_price + txn.value > self.balance:
+            raise  # Transfer value meets or exceeds account balance
 
-# NOTE: AddressAPI is a dataclass already
-class AccountAPI(AddressAPI):
-    container: "AccountContainerAPI"
+        txn = self.sign_transaction(txn)
+        return self._active_provider.send_transaction(txn)
 
-    @property
-    def alias(self) -> Optional[str]:
-        """
-        Override with whatever alias might want to use, if applicable
-        """
-        return None
+    def deploy(self, contract_type: ContractType, *args, **kwargs) -> ContractInstance:
+        kwargs["sender"] = self.address
+        c = ContractContainer(  # type: ignore
+            provider=self._active_provider,
+            contract_type=contract_type,
+        )
 
-    @abstractmethod
-    def sign_message(self, msg: SignableMessage) -> Optional[SignedMessage]:
-        ...
+        txn = c.build_deployment(*args, **kwargs)
+        txn.nonce = self.nonce
+        txn.gas_limit = self._active_provider.estimate_gas_cost(txn)
+        txn.gas_price = self._active_provider.gas_price
 
-    @abstractmethod
-    def sign_transaction(self, txn: dict) -> Optional[SignedTransaction]:
-        ...
+        txn_cost = txn.gas_limit * txn.gas_price
+        if "value" in kwargs:
+            txn_cost += kwargs["value"]
+
+        if txn_cost > self.balance:
+            raise  # Transfer value meets or exceeds account balance
```

Comment:
Is there an exception in the stack at this point?
I see this all over the code and in other situations, I have arrived there with no exceptions in the stack, causing the error to be odd.

It would be better to momentarily use `raise Exception("stuff")` and in the future, replace it with custom or more thought out errors. Unless there is something I don't understand here.

Reply:
You're totally right, was just using these as bookmarks, but probably way better to use `Exception` and customize them later

Was thinking it might be time to customize anyways, just haven't found the time

Reply:
Resolved all outstanding `raises` without `Exception`